src/Spryker/Client/RabbitMq/RabbitMqFactory.php

```diff
@@ -51,13 +64,16 @@ public function getStaticConnectionManager()
     }
 
     /**
-     * @return \Spryker\Client\RabbitMq\Model\Connection\ConnectionManagerInterface
+     * @return \Spryker\Client\RabbitMq\Model\Connection\ConnectionManager\ConnectionManagerInterface
      */
-    protected function createConnectionManager()
+    public function createConnectionManager(): ConnectionManagerInterface
     {
         $connectionManager = new ConnectionManager(
             $this->getStoreClient(),
-            $this->createConnectionFactory()
+            $this->createConnectionFactory(),
```

Comment:
I wasn't sure if all these attributes were needed for this class in the first check, after carefully looking, they are actually really used, but that's not very nice, SRP speaking, what's the difference between `ConnectionFactory` and `ConnectionCreator`, why do you need both the mapper and the filter, the filter can use the mapper or vice-versa? I think I'm confused.

Reply:
Filter and Mapper have their own responsibilities.

I guess `ConnectionFactory` and `ConnectionCreator` also have their own responsibilities `ConnectionFactory` should be stateless, and `ConnectionCreator` include logic for creating connections (Like generating `uniqueChannelId`, and caching of created connections).